### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by s4na**
It appears that our CI has stopped due to changes in Travis CI's pricing plan. I've created a PR to replace it with GitHub Actions. How about we make the switch to this instead?

refs: https://yoursunny.com/t/2020/TravisCI-credits/

---

Best regards,
s4na

### CODE REVIEW

**Comment by s4na** [comment]
In my repository, the CI passes for Ruby 2.5.3 and 2.6.3 when executed.

https://github.com/s4na/sunspot/actions/runs/5032642638

Screenshot 2023-05-20 at 23 31 07

**Comment by s4na** [comment]
@serggl 
I have created this PR because it seems the CI is not functioning in the original PR here( https://github.com/sunspot/sunspot/pull/1027 ). Could I possibly get a review?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      uses: actions/setup-java@v2
+      with:
+        java-version: '8'
+        distribution: 'adopt'
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby-version }}
+        bundler-cache: true # 'bundle install' and cache are handled automatically
+    - name: Uninstall and Install Bundler
```
Comment (serggl) [inline .github/workflows/ci.yml:40]: I think latest `setup-ruby` action allows explicit bundler versions out of the box https://github.com/ruby/setup-ruby/blob/master/action.yml#L19

**Comment by serggl** [comment]
@s4na this looks great with one small request for the bundler setup above

**Comment by serggl** [comment]
@s4na some more small suggestions, sorry :)
1. I think its safe to remove any ruby version from the build matrix which is lower than `2.5` since these ruby versions reached their EOL for more than 4 years already. Don't see much sense to still pretend to support it. This should remove most of build matrix exclusions
2. Remove ruby patch versions from build matrix, e.g. use `2.6` instead of `2.6.3`. This will test using the latest patch
3. Have you tried adding `2.7` to the build matrix? I don't recall any breaking changes in it compared to `2.6`,  hope it'll work just fine. Since this is the last pre `3.0` ruby version I would vote to test against it too.

**Comment by s4na** [comment]
@serggl 
Thank you for your review.

During my work, I realized that the master branch I was working on appears to be outdated.
After merging the latest master and running CI, it seems that the test is failing due to the code merged in this PR (https://github.com/sunspot/sunspot/pull/1026).
As a result, I've reverted it for now.

I would appreciate your review again.
Please let me know if you have any further feedback or suggestions.

## error message

```sh
An error occurred while loading ./spec/api/adapters_spec.rb.
Failure/Error:
  def boost(attr_name = nil, &block)
    @setup.add_document_boost(attr_name, &block)

ArgumentError:
  wrong number of arguments (given 2, expected 0..1)
# ./lib/sunspot/dsl/fields.rb:54:in `boost'
# ./lib/sunspot/util.rb:301:in `__proxy_method__'
# ./lib/sunspot/util.rb:293:in `method_missing'
# ./spec/mocks/comment.rb:20:in `block in <top (required)>'
# ./lib/sunspot/util.rb:257:in `instance_eval'
# ./lib/sunspot/util.rb:257:in `instance_eval_with_context'
# ./lib/sunspot/util.rb:102:in `instance_eval_or_call'
# ./lib/sunspot/setup.rb:135:in `setup'
# ./lib/sunspot/setup.rb:376:in `setup'
# ./lib/sunspot.rb:172:in `setup'
# ./spec/mocks/comment.rb:13:in `<top (required)>'
# ./spec/spec_helper.rb:8:in `require'
# ./spec/spec_helper.rb:8:in `block in <top (required)>'
# ./spec/spec_helper.rb:7:in `each'
# ./spec/spec_helper.rb:7:in `<top (required)>'
# ./spec/api/spec_helper.rb:1:in `require'
# ./spec/api/spec_helper.rb:1:in `<top (required)>'
# ./spec/api/adapters_spec.rb:1:in `require'
# ./spec/api/adapters_spec.rb:1:in `<top (required)>'
```

**Comment by serggl** [comment]
@s4na I was about to say "what a hack is happening in `sunspot/lib/sunspot/util.rb`" but you answered it. Seems like the CI was already broken at the time the PR was merged..

Does it pass for all ruby versions / gems in this way?

**Comment by s4na** [comment]
@serggl 

> Does it pass for all ruby versions / gems in this way?

Yes.

## refs
- https://github.com/s4na/sunspot/pull/2

---

Screenshot 2023-06-02 at 0 40 21

**Comment by serggl** [comment]
@s4na that's fantastic! Although I see Node 12 deprecation message on your screenshot 🤪 Will the `setup-java` version bump to v3 fix it?

**Review by serggl — APPROVED** [review]
❤️

**Pull request merged**